pinot-spi/src/main/java/org/apache/pinot/spi/config/table/H3IndexColumn.java

```diff
@@ -21,30 +21,48 @@
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
 import com.google.common.base.Preconditions;
+import java.util.ArrayList;
 import java.util.List;
+import java.util.Map;
 import org.apache.commons.collections.CollectionUtils;
 import org.apache.pinot.spi.config.BaseJsonConfig;
 
 
-public class H3IndexConfig extends BaseJsonConfig {
+public class H3IndexColumn extends BaseJsonConfig {
```

Comment:
Do we still need this class as a config class (json serializable), or this is just a helper class now? If the latter, maybe move this class to core for clarity?

Reply:
I think we can keep it as it's possible to introduce a POJO for index column and use polymorphism to support this H3indexColumn

Reply:
+1 on keeping this as a BaseJsonConfig, let's make sure we follow the same convention. should this be H3IndexConfig or H3IndexColumn or just H3Index?

Reply:
I feel `H3IndexColumn ` is better. In fact, I think it's possible to make it a subclass of `FieldConfig` by adding `@Type` per [this](https://stackoverflow.com/questions/30362446/deserialize-json-with-jackson-into-polymorphic-types-a-complete-example-is-giv).